src/main/java/com/exactpro/epfast/template/xml/TemplatesXml.java

```diff
@@ -17,67 +17,48 @@
 package com.exactpro.epfast.template.xml;
 
 import com.exactpro.epfast.template.Dictionary;
-import com.exactpro.epfast.template.Reference;
 import com.exactpro.epfast.template.Template;
 import com.exactpro.epfast.template.Templates;
 
 import javax.xml.bind.annotation.XmlAttribute;
 import javax.xml.bind.annotation.XmlElement;
 import javax.xml.bind.annotation.XmlRootElement;
+import java.util.ArrayList;
 import java.util.List;
 
 @XmlRootElement(name = "templates", namespace = NamespaceProvider.XML_NAMESPACE)
-public class TemplatesXml implements Templates, NamespaceProvider {
+public class TemplatesXml extends AbstractNamespaceProvider implements Templates {
 
-    private NamespaceProvider parentNsProvider;
-
-    private String applicationNs;
-
-    private String templateNs;
-
-    private Dictionary dictionary;
-
-    private List<Template> templates;
+    private List<Template> templates = new ArrayList<>();
 
     @Override
-    public String getApplicationNamespace() {
-        if (applicationNs != null) {
-            return applicationNs;
-        }
-        if (parentNsProvider != null) {
-            return parentNsProvider.getApplicationNamespace();
-        }
-        return Reference.DEFAULT_NAMESPACE;
+    public String getTemplateNamespace() {
```

Comment:
It isn't needed. For setters you need additional method in this class because you need to place annotation on it.